httpcore5/src/main/java/org/apache/hc/core5/util/Deadline.java

```diff
@@ -83,8 +83,11 @@ public class Deadline {
      */
     public static Deadline calculate(final long timeMillis, final TimeValue timeValue) {
         if (TimeValue.isPositive(timeValue)) {
-            // TODO handle unlikely overflow
-            final long deadline = timeMillis + timeValue.toMilliseconds();
+            final long timeToAdd = timeValue.toMilliseconds();
```

Comment:
@arturobernalg Should not this do the same, just with a bit less code. I also think `timeToAdd > 0`  and `deadline < 0` are unnecessary.

```
        if (TimeValue.isPositive(timeValue)) {
            return Deadline.fromUnixMilliseconds(timeMillis +
                    Math.min(timeValue.toMilliseconds(), Long.MAX_VALUE - timeMillis));
        }
        return Deadline.MAX_VALUE;
```

Reply:
@ok2c  Updated the implementation as suggested. Thanks for the feedback!